hazelcast-utils/pom.xml

```
@@ -0,0 +1,83 @@
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://hazelcast.com/hazelcast-community-license
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <!--
+    The purpose of this module is to collect support utility functionality that is common
+    to both client and server. The utils module should remain relatively slim; things like
+    networking should be placed in their own modules. With proper modularization we can
+    start to reduce the coupling in the code.
+
+    Functionality that is specific to clients or members should not be moved into this module.
+    -->
+
+    <name>hazelcast-utils</name>
+    <artifactId>hazelcast-utils</artifactId>
+    <description>The Hazelcast utils module.</description>
+    <packaging>jar</packaging>
+
+    <parent>
+        <groupId>com.hazelcast</groupId>
+        <artifactId>hazelcast-root</artifactId>
+        <version>5.4.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <properties>
+        <!-- Needed for CheckStyle -->
+        <main.basedir>${project.parent.basedir}</main.basedir>
+     </properties>
+
+    <build>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
+            </resource>
+        </resources>
+
+        <plugins>
+            <!-- This is needed to get rid of the Member.class. It will be replaced by the
+            right Member.class implementation from the main Hazelcast module-->
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-antrun-plugin</artifactId>
+                <version>${maven.antrun.plugin.version}</version>
+                <executions>
+                    <execution>
+                        <phase>prepare-package</phase>
+                        <goals>
+                            <goal>run</goal>
+                        </goals>
+                        <configuration>
+                            <tasks>
+                                <delete>
+                                    <fileset dir="${basedir}/target/classes/com/hazelcast/cluster"
+                                             includes="Member.class" />
+                                </delete>
+                            </tasks>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
```

I think the following would be more Maven-like:

```suggestion
          <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-jar-plugin</artifactId>
                <version>${maven.jar.plugin.version}</version>
                <configuration>
                    <excludes>
                        <exclude>com/hazelcast/cluster/Member.class</exclude>
                    </excludes>
                </configuration>
            </plugin>
```